apps/balancer-tools/src/app/poolsimulator/(components)/PoolParamsForm.tsx

```diff
@@ -55,110 +58,155 @@ const inputMapper: InputMapperType = {
       label: "Swap Fee",
       placeholder: "Enter swap fee",
       unit: "%",
+      transformFromDataToForm: (n) => (n ? n * 100 : undefined),
+      transformFromFormToData: (n) => (n ? n / 100 : undefined),
     },
     {
       name: "alpha",
       label: "Alpha",
       placeholder: "Enter alpha",
       unit: "",
+      transformFromDataToForm: (n) => n,
+      transformFromFormToData: (n) => n,
     },
     {
       name: "beta",
       label: "Beta",
       placeholder: "Enter beta",
       unit: "",
+      transformFromDataToForm: (n) => n,
+      transformFromFormToData: (n) => n,
     },
     {
       name: "lambda",
       label: "Lambda",
       placeholder: "Enter lambda",
       unit: "",
+      transformFromDataToForm: (n) => n,
+      transformFromFormToData: (n) => n,
     },
     {
       name: "c",
       label: "C",
       placeholder: "Enter c",
       unit: "",
+      transformFromDataToForm: (n) => n,
+      transformFromFormToData: (n) => n,
     },
     {
       name: "s",
       label: "S",
       placeholder: "Enter s",
       unit: "",
+      transformFromDataToForm: (n) => n,
+      transformFromFormToData: (n) => n,
     },
   ],
 };
 
-export function PoolParamsForm() {
-  const { push } = useRouter();
-  const {
-    setIsGraphLoading,
-    setInitialData,
-    setCustomData,
-    initialData,
-    poolType,
-  } = usePoolSimulator();
+export function PoolParamsForm({
+  extraOnSubmit,
+}: {
+  extraOnSubmit?: (data: AnalysisData, tabClicked: boolean) => void;
+}) {
+  const { data, setData, isCustomData } = usePoolFormContext();
 
   const form = useForm({
-    resolver: zodResolver(schemaMapper[poolType]),
-    mode: "onSubmit",
+    resolver: zodResolver(schemaMapper[data.poolType]),
+    mode: "onChange",
   });
   const {
     register,
     setValue,
-    getValues,
     clearErrors,
+    watch,
     formState: { errors },
   } = form;
 
-  const onSubmit = (data: FieldValues) => {
-    setIsGraphLoading(true);
+  const getOnSubmit = (tabClicked: boolean) => (fieldData: FieldValues) => {
+    const hasNullData = inputMapper[data.poolType].reduce(
+      (sum, input) => sum || !fieldData[input.name],
+      false
+    );
+    if (Object.keys(errors).length || hasNullData) return;
     const dataWithPoolType = {
       poolParams: Object.fromEntries(
-        inputMapper[poolType].map((input) => [input.name, data[input.name]])
+        inputMapper[data.poolType].map((input) => [
+          input.name,
+          input.transformFromFormToData(fieldData[input.name]),
+        ])
       ),
-      tokens: initialData.tokens,
-      poolType,
+      tokens: fieldData.tokens,
+      poolType: data.poolType,
     };
-
-    setInitialData(dataWithPoolType as AnalysisData);
-    setCustomData(dataWithPoolType as AnalysisData);
-    push("/poolsimulator/analysis");
+    setData(dataWithPoolType as AnalysisData);
+    extraOnSubmit?.(dataWithPoolType as AnalysisData, tabClicked);
```

Comment:
Yes! In my view, the standard behavior of this on submit is to save the data that was being modified. However, we can have to do other things depending on the context that the form is being used (such as changing the tab/page, setting the other data...). This would also be useful on the `analysis` page where we would probably have to just save the data.